### ISSUE TITLE
Implement shims for more of the Windows console APIs

### ISSUE THREAD

**Issue by thomcc** [rust-lang/miri#2652]
Miri punts on `GetConsoleMode` currently, which means that under miri, the stdlib's stdio implementation is rather different than reality. This means several things that don't work in reality will work under miri (such as writing invalid UTF-8 to stdout), and also that the [windows stdio code](https://github.com/rust-lang/rust/blob/master/library/std/src/sys/windows/stdio.rs) continues to be untested (not as worrisome as some of the sys::windows code, although it's probably the trickiest code with regards to `MaybeUninit`. But generally, it would be nice for us to align reality with what miri tests.

I took a look and I'm not sure how to do this really. It seems like much of windows IO is smoke and mirrors. So this almost certainly requires implementing a few other calls as well, such as `WriteConsoleW` and the like, although I think you can just implement console coloring as a no-op for now.

**Comment by RalfJung** [comment]
Yeah, on Windows I generally did the bare minimum needed to get the tests to pass. Many functions are barely functional stubs and certainly they are not modeling UB accurately. I'm happy to review PRs that improve Miri in this area but won't work on that myself.

So this is part of https://github.com/rust-lang/miri/issues/1537.
